spy_linux.go

```diff
@@ -32,17 +32,24 @@ func (c *pnConnIter) Next() *Connection {
 
 // cbConnections sets Connections()
 var cbConnections = func(processes bool) (ConnIter, error) {
+	// We read /proc/<pid>/net/tcp once per netns
+	netns := map[uint64]struct{}{}
 	buf := bufPool.Get().(*bytes.Buffer)
 	buf.Reset()
-	readFile(procRoot+"/net/tcp", buf)
-	readFile(procRoot+"/net/tcp6", buf)
+
 	var procs map[uint64]Proc
 	if processes {
 		var err error
-		if procs, err = walkProcPid(); err != nil {
+		if procs, err = walkProcPid(&netns, buf); err != nil {
 			return nil, err
 		}
 	}
+
+	if len(netns) == 0 {
```

Comment:
Yeah that works for me; I originally intended to do the reading in the outer functions, but it makes more sense like this.